streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -216,6 +216,8 @@ public interface UserTaskAssignmentListener {
     private AtomicLong nextScheduledRebalanceMs;
     private Queue<StreamsException> nonFatalExceptionsToHandle;
     private Time time;
+    private long missingSourceTopicTimeoutMs;
```

Comment:
The idea was to implement this for the new protocol. Check `StreamThread.handleStreamsRebalanceData` and see if you can add the logic there.